itou/job_applications/models.py

```diff
@@ -308,6 +319,20 @@ def can_download_approval_as_pdf(self):
             and self.approval.is_valid
         )
 
+    @property
+    def cancellation_delay(self):
+        delay_starts_at = self.hiring_start_at - relativedelta(days=self.CANCELLATION_DAYS_BEFORE_HIRING_STARTED)
+        delay_ends_at = self.hiring_start_at + relativedelta(days=self.CANCELLATION_DAYS_AFTER_HIRING_STARTED)
+        Delay = namedtuple("Delay", "start end")
+        return Delay(delay_starts_at, delay_ends_at)
+
+    @property
+    def can_be_cancelled(self):
+        if self.hiring_start_at:
+            cancellation_delay = self.cancellation_delay
```

Comment:
ligne pas nécéssaire imho

Reply:
`self.cancellation_delay` est une propriété, donc cela lancerait la méthode deux fois si je ne l'affectais pas à une variable. Non ?

Reply:
Correct ! TIL. 

_Python does not store the property in memory. A property in Python is just a method without (). Therefore, it will run every time you access the property. ... You can store the result of the property method, inside an attribute inside of the init method._